dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04330WorkflowTaskAddLanguageIdColumn.java

```diff
@@ -0,0 +1,275 @@
+
+        Logger.info(this, "Adding new 'language_id' column to index to 'workflow_task' table.");
+
+        try {
+
+            if (DbConnectionFactory.isMsSql() && !DbConnectionFactory.getAutoCommit()) {
+                DbConnectionFactory.setAutoCommit(true);
+            }
+
+            new DotConnect().executeStatement(getAddLanguageIdIndex());
+        } catch (SQLException e) {
+            throw new DotRuntimeException("The could not add the index to the column 'language_id' on 'workflow_task' table.", e);
+        }
+    }
+
+    private void addWorkflowTaskLanguageIdFK() {
+
+        Logger.info(this, "Adding new 'FK_workflow_task_language' FK to 'workflow_task' table.");
+
+        try {
+
+            if (DbConnectionFactory.isMsSql() && !DbConnectionFactory.getAutoCommit()) {
+                DbConnectionFactory.setAutoCommit(true);
+            }
+
+            new DotConnect().executeStatement(getAddLanguageIdForeignKey());
+        } catch (SQLException e) {
+            throw new DotRuntimeException("The could not add the FK to the column 'language_id' on 'workflow_task' table.", e);
+        }
+    }
+
+
+    private void updateAllWorkflowTaskToDefaultLanguage() throws DotDataException {
+
+        final long id = this.getDefaultLanguageId();
+
+        if (id > 0) {
+
+            Logger.info(this, "Updating all workflow_task to the default language: "
+                    + id + ", rows affected: " +
+                    new DotConnect().executeUpdate(this.getUpdateLanguageIdColumnSQL(), id));
+        } else {
+
+            Logger.warn(this, "The default language id: " + id +
+                            " is not valid, the current workflow task were not updated");
+        }
+    }
+
+    private String getAddLanguageIdForeignKey() {
+
+        final DbType dbType = DbType.getDbType(DbConnectionFactory.getDBType());
+
+        return addLanguageIdForeignKeySQLMap.getOrDefault(dbType, null);
+    }
+
+    private String getUpdateLanguageIdColumnSQL() {
+
+        final DbType dbType = DbType.getDbType(DbConnectionFactory.getDBType());
+
+        return updateLanguageIdColumnSQLMap.getOrDefault(dbType, null);
+    }
+
+
+
+    private String getAddLanguageIdIndex() {
+
+        final DbType dbType = DbType.getDbType(DbConnectionFactory.getDBType());
+
+        return addLanguageIdIndexSQLMap.getOrDefault(dbType, null);
+    }
+
+    private String getAddLanguageIdColumnSQL() {
+
+        final DbType dbType = DbType.getDbType(DbConnectionFactory.getDBType());
+
+        return addLanguageIdColumnSQLMap.getOrDefault(dbType, null);
+    }
+
+    private String getSelectLanguageIdColumnSQL() {
+
+        final DbType dbType = DbType.getDbType(DbConnectionFactory.getDBType());
+
+        return selectLanguageIdColumnSQLMap.getOrDefault(dbType, null);
+    }
+
+    private long getDefaultLanguageId() throws DotDataException {
+
+        final String languageCode = Config.getStringProperty("DEFAULT_LANGUAGE_CODE");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getStringProperty"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)